include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,531 @@
+
+      // change the layout from the vertical layout to the top horizontal one
+      _chart.parent = rowChartsLayout;
+  }
+
+  function onSubscribe(_Id, _topic, _path)
+  {
+      PlottingIface.subscribe(_Id, _topic, _path);
+  }
+  function onUnSubscribe(_Id, _topic, _path)
+  {
+      PlottingIface.unsubscribe(_Id, _topic, _path);
+  }
+  function onComponentSubscribe(entity, typeId, type, attribute, Id)
+  {
+      PlottingIface.onComponentSubscribe(entity, typeId, type, attribute, Id);
+  }
+  function onComponentUnSubscribe(entity, typeId, attribute, Id)
+  {
+      PlottingIface.onComponentUnSubscribe(entity, typeId, attribute, Id);
+  }
+  function onClicked(Id)
+  {
+      // if many charts mode & the selected chart is in the horizontal layout
+      if (multiChartsMode && charts[Id].multiChartsMode)
+      {
+          // ======= main chart =========
+          // change the main charts properties to fit in the horizontal layout
+          charts[mainChartID].width = 200;
+          charts[mainChartID].height = Qt.binding( function() {return rowCharts.height * 0.9} );
+          charts[mainChartID].y = (rowCharts.height - charts[mainChartID].height)/2;
+          charts[mainChartID].multiChartsMode = true;
+          charts[mainChartID].fillPlotInOrOut();
+
+          // swap the main chart with the position of the clicked chart
+          charts[mainChartID].parent = rowChartsLayout;
+
+          // ======= swapped chart =======
+          charts[Id].parent = chartsLayout;
+          charts[Id].x = charts[mainChartID].x;
+          charts[Id].y = 0;
+          charts[Id].height = Qt.binding( function() {return chartsLayout.height / chartsLayout.heightFactor});
+          charts[Id].width = Qt.binding( function() {return chartsLayout.width});
+          charts[Id].multiChartsMode = false;
+          charts[Id].fillPlotInOrOut();
+          charts[Id].setChartOpacity(1);
+
+          mainChartID = Id;
+      }
+  }
+
+  function handlePlot(_chart, _fieldID, _x, _y)
+  {
+    charts[_chart].appendPoint(_fieldID, _x, _y);
+  }
+
+  function moveCharts()
+  {
+    Object.keys(charts).forEach(function(key) {
+        charts[key].moveChart();
+    });
+  }
+
+  Connections {
+      target: PlottingIface
+      onPlot : handlePlot(_chart, _fieldID, _x, _y);
+      onMoveChart : moveCharts();
+  }
+
+  Component.onCompleted: {
+      addChart();
+  }
+
+  property var window: null
+
+  ToolButton {
+      id: open
+      width: 40;
+      height: 40
+      anchors.right: parent.right
+      anchors.top: addBtn.bottom
+      anchors.margins: 10
+      onHoveredChanged: (opacity === 1) ? opacity = 0.8 : opacity = 1;
+
+      icon.width: parent.width
+      icon.height: parent.height
+      icon.color: (Material.theme == Material.Light) ? Material.color(Material.Grey, Material.Shade700)
+                                                     : Material.color(Material.Grey, Material.Shade400)
+
+      icon.source: "images/export_icon.png"
```

Comment:
I'm experiencing issue loading `icon` with Qt 5.9
```
file:///home/claire/gsoc_dome_ws/src/ign-gui/include/ignition/gui/qml/PlottingInterface.qml:259 Cannot assign to non-existent property "icon"
```
With a bit of research, I found some people say `icon` doesn't work prior to Qt5.10 https://github.com/moonlight-stream/moonlight-qt/issues/53#issuecomment-417181817. I was wondering if this is another modern feature that needs to be adjusted

Reply:
here is a potential solution https://github.com/moonlight-stream/moonlight-qt/commit/0467c2f893cb35b7cab941d7a5d258f1e5731730#diff-81df80a5e56b87fa3890edbb06254763R75 

They used the icon image instead to make it compatible with Qt5.9